docs/guide/CSS.md

```diff
@@ -0,0 +1,395 @@
+# Textual CSS
+
+Textual uses CSS to apply style to widgets. If you have any exposure to web development you will have encountered CSS, but don't worry if you haven't: this section will get you up to speed.
+
+## Stylesheets
+
+CSS stands for _Cascading Stylesheets_. A stylesheet is a list of styles and rules about how those styles should be applied to a page. In the case of Textual, the stylesheet applies styles to widgets but otherwise it is the same idea.
+
+!!! note
+
+    Depending on what you want to build with Textual, you may not need to learn Textual CSS at all. Widgets are packaged with CSS styles so apps with exclusively pre-built widgets may not need any additional CSS.
+
+Textual CSS defines a set of rules which apply visual _styles_ to your application and widgets. These style can customize a large variety of visual settings, such as color, border, size, alignment; and more dynamic features such as animation and hover effects. As powerful as it is, CSS in Textual is quite straightforward.
+
+CSS is typically stored in an external file with the extension `.css` alongside your Python code.
+
+Let's look at some Textual CSS.
+
+```sass
+Header {
+  dock: top;
+  height: 3;
+  content-align: center middle;
+  background: blue;
+  color: white;
+}
+```
+
+This is an example of a CSS _rule set_. There may be many such sections in any given CSS file.
+
+Let's break this CSS code down a bit.
+
+```css hl_lines="1"
+Header {
+  dock: top;
+  height: 3;
+  content-align: center middle;
+  background: blue;
+  color: white;
+}
+```
+
+The first line is a _selector_ which tells Textual which Widget(s) to modify. In the above example, the styles will be applied to a widget defined in the Python class `Header`.
+
+```css hl_lines="2 3 4 5 6"
+Header {
+  dock: top;
+  height: 3;
+  content-align: center middle;
+  background: blue;
+  color: white;
+}
+```
+
+The lines inside the curly braces contains CSS _rules_, which consist of a rule name and rule value separated by a colon and ending in a semi-colon. Such rules are typically written one per line, but you could add additional rules as long as they are separated by semi-colons.
+
+The first rule in the above example reads `"dock: top;"`. The rule name is `dock` which tells Textual to place the widget on a edge of the screen. The text after the colon is `top` which tells Textual to dock to the _top_ of the screen. Other valid values for dock are "right", "bottom", or "left"; but `top` is naturally appropriate for a header.
+
+You may be able to guess what some of the the other rules do. We will cover those later.
+
+## The DOM
+
+The DOM, or _Document Object Model_, is a term borrowed from the web world. Textual doesn't use documents but the term has stuck. In Textual CSS, the DOM is a an arrangement of widgets that forms a tree of sorts.
```

Comment:
Not a (personal) fan of “forms a tree of sorts” unless this is the kind of tone you want in the docs.

 - ... the DOM is an arrangement of widgets you can visualise in a tree-like hierarchical structure.
 - ... the DOM is the (tree-like) hierarchy that establishes the relationships between the widgets.
 - ...?

Reply:
I'll rephrase. Needs to be a little informal though, for novices devs without a background in computer science.

Reply:
It is hard to strike a balance between informal and precise, but I'm sure it is attainable.